### PR TITLE
Convert to SnakeYAML 2.0; add test

### PR DESCRIPTION
**Pull request by tjquinno**
See description here: #6384

### CODE REVIEW

File: tests/integration/gh-5792/.dockerignore [hunk 1]
```diff
@@ -0,0 +1 @@
+target/*
```
Comment (barchetta) [inline tests/integration/gh-5792/.dockerignore:1]: Is this file needed?

File: tests/integration/gh-5792/app.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#
```
Comment (barchetta) [inline tests/integration/gh-5792/app.yaml:1]: Is this file needed?

File: tests/integration/gh-5792/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply tests/integration/gh-5792/Dockerfile:1]: I just added all the files that the archetype generated. Given that the test does not use docker or native image, we don't need them. I'll remove them.

**Review by barchetta — APPROVED** [review]
LGTM
